src/main/java/uk/gov/hmcts/sptribs/systemupdate/schedule/wa/SystemTriggerCompleteHearingOutcomeTask.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.hmcts.sptribs.systemupdate.schedule.wa;
+
+import lombok.extern.slf4j.Slf4j;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.idam.client.models.User;
+import uk.gov.hmcts.sptribs.idam.IdamService;
+import uk.gov.hmcts.sptribs.systemupdate.service.CcdConflictException;
+import uk.gov.hmcts.sptribs.systemupdate.service.CcdManagementException;
+import uk.gov.hmcts.sptribs.systemupdate.service.CcdSearchCaseException;
+import uk.gov.hmcts.sptribs.systemupdate.service.CcdSearchService;
+import uk.gov.hmcts.sptribs.systemupdate.service.CcdUpdateService;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
+import static org.elasticsearch.index.query.QueryBuilders.matchQuery;
+import static org.elasticsearch.index.query.QueryBuilders.rangeQuery;
+import static uk.gov.hmcts.sptribs.systemupdate.event.SystemTriggerCompleteHearingOutcome.SYSTEM_TRIGGER_COMPLETE_HEARING_OUTCOME;
+
+@Component
+@Slf4j
+public class SystemTriggerCompleteHearingOutcomeTask implements Runnable {
+
+    private final CcdSearchService ccdSearchService;
+
+    private final CcdUpdateService ccdUpdateService;
+
+    private final IdamService idamService;
+
+    private final AuthTokenGenerator authTokenGenerator;
+
+    @Autowired
+    public SystemTriggerCompleteHearingOutcomeTask(CcdSearchService ccdSearchService, CcdUpdateService ccdUpdateService,
+                                                   IdamService idamService, AuthTokenGenerator authTokenGenerator) {
+        this.ccdSearchService = ccdSearchService;
+        this.ccdUpdateService = ccdUpdateService;
+        this.idamService = idamService;
+        this.authTokenGenerator = authTokenGenerator;
+    }
+
+    @Override
+    public void run() {
+        final User user = idamService.retrieveSystemUpdateUserDetails();
+        final String serviceAuth = authTokenGenerator.generate();
+
+        try {
+            final BoolQueryBuilder query = boolQuery()
```

Comment:
We need a flag for this one so that it isn't picked up by the cron job more than once

Reply:
If the cron job runs once per day, I think it shouldn't be picked up again unless the hearing is postponed?

Reply:
OK cool - one to bear in mind when testing in lower envs when we may run the cron more frequently